packages/thirdweb/src/utils/any-evm/deploy-metadata.ts

```diff
@@ -22,26 +23,30 @@
 export async function fetchDeployMetadata(
   options: FetchDeployMetadataOptions,
 ): Promise<FetchDeployMetadataResult> {
+  const isZksolc = options.compilerType === "zksolc";
   const rawMeta: RawCompilerMetadata = await download({
     uri: options.uri,
     client: options.client,
   }).then((r) => r.json());
-  // TODO: proper handling of different compiler metadata types
-  const metadataUri =
-    rawMeta.compilers?.zksolc?.length > 0 && rawMeta.name.endsWith("_ZkSync")
-      ? rawMeta.compilers.zksolc[0].metadataUri
-      : rawMeta.metadataUri;
-  const bytecodeUri =
-    rawMeta.compilers?.zksolc?.length > 0 && rawMeta.name.endsWith("_ZkSync")
-      ? rawMeta.compilers.zksolc[0].bytecodeUri
-      : rawMeta.bytecodeUri;
+
+  if (isZksolc && rawMeta.compilers?.zksolc.length === 0) {
+    throw new Error("Invalid compiler type");
```

Comment:
would change this error msg to something like `no zksolc metadat found for contract: ${rawMeta.name}`